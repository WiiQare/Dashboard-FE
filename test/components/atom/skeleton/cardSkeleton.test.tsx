import React from 'react';
import { render } from '@testing-library/react';
import CardSkeleton from '../../../../src/components/atom/skeleton/cardSkeleton';

describe('CardSkeleton Component', () => {
  it('renders the correct number of skeleton cards', () => {
    const numberOfCards = 3;
    const { container } = render(<CardSkeleton number={numberOfCards} />);

    const skeletonCards = container.querySelectorAll('.grid-cols-1');

    expect(skeletonCards).toMatchSnapshot();
  });
});
