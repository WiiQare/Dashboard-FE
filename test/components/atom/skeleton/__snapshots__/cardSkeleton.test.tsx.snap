// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkeleton Component renders the correct number of skeleton cards 1`] = `
NodeList [
  <div
    class="grid cards grid-cols-1 gap-2 md:grid-cols-2 md:gap-6 xl:grid-cols-5 2xl:gap-7.5 p-[23px]"
  >
    <div
      class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
    >
      <div
        class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
      >
        <div
          class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-16 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-24 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-36 h-4 bg-gray-300 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
    >
      <div
        class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
      >
        <div
          class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-16 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-24 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-36 h-4 bg-gray-300 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
    >
      <div
        class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
      >
        <div
          class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-16 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-24 h-4 bg-gray-300 rounded-md "
          />
          <div
            class="w-36 h-4 bg-gray-300 rounded-md "
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
